usefulUtilities/voteApp/appResources/voteApp.js

```diff
@@ -0,0 +1,1022 @@
+"use strict";
+
+//  
+//  Vote App | voteApp.js
+//
+//  Created by Robin Wilson 2018-11-12
+//
+//  Using Example Vue App Created by Milad Nazeri on 2018-10-11
+//  Copyright 2016 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* globals Tablet, Script, HMD, Controller, AccountServices, Menu */
+
+(function () { // BEGIN LOCAL_SCOPE
+
+    // Modules
+    var AppUi = Script.require('appUi'),
+        GOOGLE_SCRIPTS_URL = Script.require(Script.resolvePath("./resources/secrets/secrets.js")).googleScriptsUrl,
+        URL = Script.resolvePath("./resources/voteApp.html?v12344"),
+        CONFIG = Script.require(Script.resolvePath("./resources/config.js?v12344"));
+
+    // Configurable variables
+    var EVENT_DATE = CONFIG.EVENT_DATE,
+        EVENT_NAME = CONFIG.EVENT_NAME;
+
+    // Unload app variables
+    // Example: UTC month 0 = January ; month 10 = sNovember
+    // year, month, day, hour, minutes
+    var UNLOAD_DATE = CONFIG.UNLOAD_DATE,
+        UNLOAD_TIME = 10000; // show app will unload screen for 10 seconds
+    
+    // App variables
+    var BUTTON_NAME = "VOTE",
+        VOTE_APP_NAME = "voteApp.js";
+
+    // App status variables
+    var firstLoad = true,
+        setupNotLoggedIn = false,
+        FIRST_LOAD_TIMEOUT = 500,
+        REQUEST_READY_STATE = 4;
+
+    // Domains vs Zone visited
+    var DOMAINS_ENABLED = true; // enable domains UI / domain visited checks
+    
+    // Zone visited
+    var intervalCheckVisited,
+        TIME_CHECK = 1000,
+        HAS_VISITED_ZONE_CHECK = false, // if there's a zone to visit
+        GOTO_ADDRESS = "thespot//-26.8224,-11.1442,-67.663/0,0.068694,0,0.997638";
+
+    // Web event types from UI
+    var EVENT_BRIDGE_OPEN_MESSAGE = CONFIG.EVENT_BRIDGE_OPEN_MESSAGE,
+        GOTO_LOCATION = CONFIG.GOTO_LOCATION,
+        GOTO_DOMAIN = CONFIG.GOTO_DOMAIN,
+        VOTE_AVATAR = CONFIG.VOTE_AVATAR,
+        VOTE_DOMAIN = CONFIG.VOTE_DOMAIN;
+
+    // Google scripts type events
+    var VOTE_GOOGLE = "vote",
+        GET_INFO_GOOGLE = "getInfo";
+
+    // Vue UI update event
+    var UPDATE_UI = EVENT_NAME + "_update_ui"; // !important must match Tablet.js
+    
+    // Static strings
+    var DOMAIN = "domain";
+    var AVATAR = "avatar";
+    
+    var DEBUG = true;
+
+    // Local Settings.setValue/getValue key
+    var VOTE_APP_SETTINGS_NAME = EVENT_NAME + "_voteApp"; // "Futvrelands_11_17_2018_voteApp"
+    
+    var DEFAULT_VOTE_APP_SETTINGS = {
+        visited: false,
+        domains: [],
+        voted: {
+            avatar: "", domain: ""
+        }
+    };
+
+    // UI variables
+    var ui;
+    var dataStore = {
```

Comment:
what is the significance of the concept of a `dataStore` rather than nakedly storing these as globals?

Reply:
dataStore is managed in a central place to be passed into Vue as the data to render:

    // Vue UI update event
    var UPDATE_UI = CONFIG.UPDATE_UI;

    function updateUI() {
        var messageObject = {
            type: UPDATE_UI,
            **value: dataStore**
        };
        ui.sendToHtml(messageObject);
    }